inc/templates.php

```diff
@@ -274,3 +279,159 @@ function convert_blocks_to_components( array $blocks ): array {
 
 	return $components;
 }
+
+/**
+ * Recursively iterate on a component tree.
+ *
+ * @param array $components Array of components.
+ * @return array
+ */
+function traverse_components( array $components ): array {
+
+	foreach ( $components as $index => &$component ) {
+
+		// This allows us to set text nodes.
+		if ( is_string( $component ) ) {
+			continue;
+		}
+
+		// Ensure we have all the right field and types.
+		$component = ensure_component_fields_exist( $component );
```

Comment:
I wonder if instantiating a `WP_Component` object here would allow us to avoid needing to juggle validation by handling it inside the component object?

Reply:
Agreed, we have a ticket for this.